samples/matter/lock/src/access/access_data_types.h

```diff
@@ -117,8 +117,71 @@ struct Credential {
 	 */
 	CHIP_ERROR Deserialize(const void *buff, size_t buffSize);
 };
+/* A base class for specific schedule types that implements the common schedule serialization. */
```

Comment:
We discussed using templates here previously on my PR with the first support and decided to not make a template here, and I can see that you got back to this approach. What happened that you decided to make a template here? We didn't want this approach after a discussion with @kkasperczyk-no. TBH, I like this approach and would like to have it, but I can see some inconsistencies here with our previous decisions :)

Reply:
To be honest I cannot recall such discussion, so it's hard to me to comment on that. If you both are ok with this one, as it solves some problem in generic way, maybe it's ok to use template.

Reply:
Ahh right, I recall this discussion but not sure it it was regarding this particular bit. Anyway, during fixing of the recent serialization bug I found it way more convenient to change the code (which does exactly the same) in only one place instead of deploying the changes all over the place. Thus it came natural to me that we should unify these methods. Our last discussion may have lead to a different conclusion but in practice it turned out we were better off having the boiler plate gone IMO.

Reply:
Ok, so let's go with your approach :)